pyscf/lib/vhf/optimizer.c

```diff
@@ -169,6 +166,129 @@ int CVHFnrs8_vk_prescreen(int *shls, CVHFOpt *opt,
             || (  dm_cond[i*n+l]*qijkl > direct_scf_cutoff));
 }
 
+int CVHFnrs8_vj_prescreen_block(CVHFOpt *opt, int *ishls, int *jshls, int *kshls, int *lshls)
+{
+        int i0 = ishls[0];
+        int j0 = jshls[0];
+        int k0 = kshls[0];
+        int l0 = lshls[0];
+        int i1 = ishls[1];
+        int j1 = jshls[1];
+        int k1 = kshls[1];
+        int l1 = lshls[1];
+        int i, j, k, l;
+        size_t n = opt->nbas;
+        double *q_cond = opt->q_cond;
+        double *dm_cond = opt->dm_cond;
+        double direct_scf_cutoff = opt->direct_scf_cutoff;
+        double rho, cutoff;
+        rho = 0;
+        for (j = j0; j < j1; j++) {
+#pragma GCC ivdep
+        for (i = i0; i < i1; i++) {
+                rho += dm_cond[j*n+i] * q_cond[j*n+i];
+        } }
+
+        if (rho != 0) {
+                cutoff = 4 * direct_scf_cutoff / fabs(rho);
+                for (l = l0; l < l1; l++) {
+                for (k = k0; k < k1; k++) {
+                        if (q_cond[l*n+k] > cutoff) {
+                                return 1;
+                        }
+                } }
+        }
+
+        rho = 0;
+        for (l = l0; l < l1; l++) {
+#pragma GCC ivdep
```

Comment:
I suggest `#pragma omp simd`

Reply:
Seems a new clause introduced by omp new versions. Not sure how compiler supports this statement.